### PR TITLE
hardening/messageLengthRequired

### PR DESCRIPTION
**Pull request by rbl219**
Resolved issue #1203

Changing error message length required.

### CODE REVIEW

File: src/lib/rest/rest.cpp [hunk 1]
```diff
@@ -1025,7 +1025,7 @@ static int connectionTreat
   // 3. Finally, serve the request (unless an error has occurred)
   if (((ciP->verb == POST) || (ciP->verb == PUT)) && (ciP->httpHeaders.contentLength == 0) && (strncasecmp(ciP->url.c_str(), "/log/", 5) != 0))
   {
-    std::string errorMsg = restErrorReplyGet(ciP, ciP->outFormat, "", url, SccLengthRequired, "Zero/No Content-Length in PUT/POST request");
+    std::string errorMsg = restErrorReplyGet(ciP, ciP->outFormat, "", url, SccLengthRequired, "Zero/No Content-Length in PUT/POST/PATCH request");
```
Comment (fgalan) [inline src/lib/rest/rest.cpp]: Have you run the functional tests (i.e. `make ft`)  to check that all the existing tests are ok? Typically, when output changes, some test need litle adjusts.

File: src/lib/rest/rest.cpp [hunk 2]
```diff
@@ -1025,7 +1025,7 @@ static int connectionTreat
   // 3. Finally, serve the request (unless an error has occurred)
```
Comment (fgalan) [inline src/lib/rest/rest.cpp]: CHANGES_NEXT_RELEASE entry about the fix should be added.

Comment (rbl219) [inline src/lib/rest/rest.cpp]: No sorry, have only run unit tests. I check functional tests now !!

File: src/lib/rest/rest.cpp (hunk 1 above)
Comment (rbl219) [inline src/lib/rest/rest.cpp]: Fix in commit 8b3ffb8

File: src/lib/rest/rest.cpp (hunk 2 above)
Comment (rbl219) [inline src/lib/rest/rest.cpp]: Fix in commit 8b3ffb8

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -32,3 +32,4 @@ Fix:  Removed the check of forbidden characters for the values of the URI parame
 Add:  GET /v2/types (Issue #1027)
 Fix:  Made broker respond with 400 Bad Request on encountering invalid items in URI parameter 'options' (Issue #1169)
 Add:  text/plain encoding for GET /v2/entities/<entityId>/attrs/<attrName>/value operation (both Accept header and ?options=text) (Issue #1179)
+Fix:  Fixed a bug about HTTP Content-Length required (Issue #1203)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: I'd suggest to be more specific, e.g.:

"Fix: detail message for HTTP Content-Lenght required error (Issue #1203)"

Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Fixed in 44ebe21.

**Comment by fgalan** [comment]
LGTM (but waiting for 0.24.0 releasing before merging, as develop is in frozen state right now)

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**